File: Swift-C++-IR-Comparison/test/structsAndClasses/Swift/ClassAssignmentVarPointer.swift
//TODO: Learning - it makes a huge difference in the generated IR, wether
// the pointer 'copied' is defined as a let constant or a variable!

@main
struct MyMain {

    static func main() {
        let vm = VideoMode()
        var copied = vm
        copied.interlaced = true
        copied.frameRate = 2.7
    }
}
class VideoMode {
    var interlaced = false
    var frameRate = 0.0
    var name: String = "My Mode"
}

File: Swift-C++-Testsuite/test/structsAndClasses/C++/Classes/ClassAssignmentVarPointer.cpp
 
#include <iostream> 

class VideoMode {
    public:
    bool interlaced = false;
    double frameRate = 0.0;
    std::string name = "MyMode";
};

int main(){
    VideoMode vm;
    VideoMode copied = vm;
    copied.interlaced = true;
    copied.frameRate = 2.7;
    return 0;
}